src/vm/methodtablebuilder.cpp

```diff
@@ -5094,6 +5094,19 @@ MethodTableBuilder::InitNewMethodDesc(
         {
             pNewMD->SetIsJitIntrinsic();
         }
+
+        // All the funtions in System.Runtime.Intrinsics.X86 are hardware intrinsics.
+        // We specially treat them here to reduce the disk footprint of mscorlib.
+        LPCUTF8 className;
+        LPCUTF8 nameSpace;
+
+        HRESULT hrns = GetMDImport()->GetNameOfTypeDef(bmtInternal->pType->GetTypeDefToken(), &className, &nameSpace);
```

Comment:
It would be better to do this check once per type, not once per each method.

Reply:
Also, it can be under `defined(_TARGET_X86_) || defined(_TARGET_AMD64_)` ifdef.

Reply:
Thanks, will do.